api/src/main/java/org/pmiops/workbench/actionaudit/adapters/AuthDomainAuditAdapter.java

```diff
@@ -0,0 +1,5 @@
+package org.pmiops.workbench.actionaudit.adapters;
+
+public interface AuthDomainAuditAdapter {
```

Comment:
I've been looking for a good place to document these. There are several such adapters, but they don't inherit from anything. I even made an empty interface just to have somewhere to put a comment, but pulled that out after I couldn't think of anything useful to hang on the interface.

It's easier to see the intent of the pattern with the `WorkspaceAuditAdapter`. These little services provide a way to instrument controllers and services with minimal footprint. You just call `fireCreate(Workapace w, long id)` or something and the adapter is in charge of building the ActionAuditEvents and passing them along. In some cases, this removes a good deal of unwieldy code that would have to go in the controller or service.

Additionally, these services are insulated from the real Cloud Logging machinery, as they just pass opaque data class instances down to that service.

Reply:
Perhaps a README.md in this directory, or some other readme

Reply:
Looking at it more, also wondering if `Adapter` is the right name for these interfaces. Not clear to me what its "adapting". I feel like I get just as much context from a shorter name like `WorkspaceAuditer`.

Reply:
Yeah, I'm not married to that name. Typically an adapter connects something we write to something somebody else wrote. Maybe `AuthDomainAuditManager` or something? With the understanding that a manager is below the service level. Or event `AuthDomainAuditService`.

I'd prefer to do that rename on a follow-on PR since this one is already bloated.